swayfmt/tests/mod.rs

```diff
@@ -1716,3 +1718,175 @@ impl OrdEq for u256 {}
 "#,
     );
 }
+
+#[test]
+fn chained_methods_1() {
+    check(
+        r#"
+library;
+
+fn test() {
+    fuelcoin.really_long_field.other_really_long_field.foo().bar().baz.quux().yet_another_call().to_go_above_max_line_length();
+}
+        "#,
+        r#"library;
+
+fn test() {
+    fuelcoin
+        .really_long_field
+        .other_really_long_field
+        .foo()
+        .bar()
+        .baz
+        .quux()
+        .yet_another_call()
+        .to_go_above_max_line_length();
+}
+"#,
+    );
+}
+
+#[test]
+fn chained_methods_2() {
+    check(
+        r#"
+library;
+
+fn test() {
+    fuelcoin.really_long_field.other_really_long_field.foo().bar().baz.quux([1,2]).yet_another_call([1, 2, 3, 4, 6, 7, 7, 8, 8, 9, 9, 9, 19, 1123123, 12312, 312, 312, 3123, 12,31, 44],[1,2], true).to_go_above_max_line_length();ping();
+}
+        "#,
+        r#"library;
+
+fn test() {
+    fuelcoin
+        .really_long_field
+        .other_really_long_field
+        .foo()
+        .bar()
+        .baz
+        .quux([1, 2])
+        .yet_another_call(
+            [
+                1, 2, 3, 4, 6, 7, 7, 8, 8, 9, 9, 9, 19, 1123123,
+                12312, 312, 312, 3123, 12, 31, 44,
+            ],
+            [1, 2],
+            true,
+        )
+        .to_go_above_max_line_length();
+    ping();
+}
+"#,
+    );
+}
+
+#[test]
+fn chained_methods_3() {
+    check(
+        r#"
+library;
+
+fn test() {
+    fuelcoin.really_long_field.other_really_long_field.foo().bar().baz.quux([1,2]).yet_another_call(1, 2, 3, 4, 6, 7, 7, 8, 8, 9, 9, 9, 19, 1123123, 12312, 312, 312, 3123, 12,31, 44,[1,2], true).to_go_above_max_line_length();
+}
+        "#,
+        r#"library;
+
+fn test() {
+    fuelcoin
+        .really_long_field
+        .other_really_long_field
+        .foo()
+        .bar()
+        .baz
+        .quux([1, 2])
+        .yet_another_call(
+            1,
+            2,
+            3,
+            4,
+            6,
+            7,
+            7,
+            8,
+            8,
+            9,
+            9,
+            9,
+            19,
+            1123123,
+            12312,
+            312,
+            312,
+            3123,
+            12,
+            31,
+            44,
+            [1, 2],
+            true,
```

Comment:
For now this is okay, but should probably emit a warning, eg Rust emits a warning for too many function parameters.

This shouldn't be emitted in the formatter though, and I don't think any users would do anything like this.

would you mind making the number of function parameters here something more reasonable? If the goal is to make it format into multiline for just the method call then I'm sure there is a better approach that won't eat up so many lines.